### PR TITLE
Added a BillboardMesh example

### PR DESCRIPTION
**Pull request by mstrater**

### CODE REVIEW

File: examples/js/objects/BillboardMesh.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+function rotateMesh( mesh, towardPoint, up ) {
+
+	mesh.up.copy( up );
+	mesh.lookAt( towardPoint );
+
+}
+
+function rotateAxialConstrainedMesh( mesh, towardPoint, axis ) {
+
+	var meshPosition = mesh.position.clone();
```
Comment (WestLangley) [inline examples/js/objects/BillboardMesh.js]: Cloning is not required, I believe.

File: examples/js/objects/BillboardMesh.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	var look = new THREE.Vector3();
```
Comment (WestLangley) [inline examples/js/objects/BillboardMesh.js]: Use a closure to avoid "new" in this method.

File: examples/webgl_billboardmesh.html [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+				controls.addEventListener( 'change', render );
+
+				scene = new THREE.Scene();
+				scene.fog = new THREE.FogExp2( 0x00000f, 0.001 );
+
+				// planet
+
+				var radius = 200;
+				var sphereGeometry = new THREE.SphereGeometry( radius, 32, 32 );
+				var sphereMaterial = new THREE.MeshPhongMaterial( { bumpMap: textures.planet, bumpScale: 10, color: 0x333344, overdraw: 0.5 } );
```
Comment (WestLangley) [inline examples/webgl_billboardmesh.html]: `overdraw` is for `CanvasRenderer` only

File: examples/webgl_billboardmesh.html [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+					theta = 2 * Math.PI * u;
+					phi = Math.acos( 2 * v - 1 );
+					surfacePoint = new THREE.Vector3( radius * Math.cos( theta ) * Math.sin( phi ), radius * Math.sin( theta ) * Math.sin( phi ), radius * Math.cos( phi ) );
+					orthogonalAxis = surfacePoint.clone();
+					orthogonalAxis.normalize();
+
+					// create an axially constrained billboard for the alien body
+					billboardBodyPosition = surfacePoint.clone().add( orthogonalAxis.clone().multiplyScalar( 20 ) );
+					alienBodyBillboard = new THREE.BillboardMesh( alienBodyGeom, alienBodyMaterial, camera.position, orthogonalAxis, true );
+					alienBodyBillboard.position.set( billboardBodyPosition.x, billboardBodyPosition.y, billboardBodyPosition.z );
```
Comment (WestLangley) [inline examples/webgl_billboardmesh.html]: `alienBodyBillboard.position.copy( billboardBodyPosition )`

File: examples/webgl_billboardmesh.html [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+					billboards.push( alienBodyBillboard );
+					scene.add( alienBodyBillboard );
+
+					// create an unconstrained billboard for the alien head (always directly faces the camera)
+					billboardHeadPosition = surfacePoint.clone().add( orthogonalAxis.clone().multiplyScalar( 75 ) );
+					alienHeadBillboard = new THREE.BillboardMesh( alienHeadGeom, alienHeadMaterial, camera.position, orthogonalAxis, false );
+					alienHeadBillboard.position.set( billboardHeadPosition.x, billboardHeadPosition.y, billboardHeadPosition.z );
```
Comment (WestLangley) [inline examples/webgl_billboardmesh.html]: `copy()`

**Comment by WestLangley** [comment]
The article you referenced explains the different types of billboarding you are implementing. It would be nice to elaborate a bit in the code header -- and provide the doc files.

File: examples/js/objects/BillboardMesh.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
+/**
+ * @author Max Strater, Autodesk, Inc. / http://www.autodesk.com/
+ *
+ * A billboard mesh that faces a specified point, optionally constrained to an axis
+ * Based on http://nehe.gamedev.net/article/billboarding_how_to/18011/
+ */
+
+// expects a normalized axis
+THREE.BillboardMesh = function ( geometry, material, point, normalizedAxis, constrainedToAxis ) {
+
```
Comment (WestLangley) [inline examples/js/objects/BillboardMesh.js]: Yes, An axis in three.js is always assumed to be normalized. You can call the parameter `axis`, and keep your comment for clarity -- as we have done elsewhere in the library.

**Comment by mstrater** [comment]
Should I make any more changes?

File: examples/js/objects/BillboardMesh.js [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+		mesh.lookAt( towardPoint );
+
+	}
+
+	var look = new THREE.Vector3();
+	var right = new THREE.Vector3();
+
+	function rotateAxialConstrainedMesh( mesh, towardPoint, axis ) {
+
+		var meshPosition = mesh.position.clone();
```
Comment (WestLangley) [inline examples/js/objects/BillboardMesh.js]: Why `clone()`?

File: examples/js/objects/BillboardMesh.js [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+		look.subVectors( towardPoint, meshPosition );
+		right.crossVectors( axis, look ).normalize();
+		look.crossVectors( right, axis ).normalize();
+		mesh.matrix.set(
+			right.x, axis.x, look.x, meshPosition.x,
+			right.y, axis.y, look.y, meshPosition.y,
+			right.z, axis.z, look.z, meshPosition.z,
+			0,       0,      0,      1
+		);
+		mesh.matrix.decompose( mesh.position, mesh.quaternion, mesh.scale );
```
Comment (WestLangley) [inline examples/js/objects/BillboardMesh.js]: By setting the matrix directly, your are clobbering any scale that may have been applied to the object. The position is not changing -- nor is the scale. All you really need to do is set the rotation/quaternion directly.

I do like your demo, but setting the matrix directly here violates the three.js paradigm.

**Comment by threejsworker** [comment]
The examples of this pullrequest are now built and visible in threejsworker. To view them, go to the following link: 

 http://threejsworker.com/viewpullrequest.html#6484

**Pull request closed without merging**